libs/libc/stdio/lib_libfilesem.c

```diff
@@ -62,36 +59,11 @@ void lib_sem_initialize(FAR struct file_struct *stream)
 
 void lib_take_semaphore(FAR struct file_struct *stream)
 {
-  pid_t my_pid = getpid();
   int ret;
 
-  /* Do I already have the semaphore? */
-
-  if (stream->fs_holder == my_pid)
-    {
-      /* Yes, just increment the number of references that I have */
-
-      stream->fs_counts++;
-    }
-  else
+  while ((ret = nxrmutex_lock(&stream->fs_lock)) < 0)
```

Comment:
let's remove the while loop

Reply:
We need to check for `ECANCELED`. This could be placed inside the lock